iaf-aspose/src/main/java/nl/nn/adapterframework/extensions/aspose/services/conv/impl/convertors/PdfAttachmentUtil.java

```diff
@@ -0,0 +1,176 @@
+import nl.nn.adapterframework.extensions.aspose.services.util.StringsUtil;
+
+/**
+ * This class will combine seperate pdf files to a single pdf with attachments.
+ * None existing files in a CisConversionResult will be skipped!
+ * 
+ * TODO: Refactor this class
+ */
+public class PdfAttachmentUtil {
+
+	private static final Logger LOGGER = Logger.getLogger(PdfAttachmentUtil.class);
+
+	private List<CisConversionResult> cisConversionResultList;
+
+	private Document pdfDocument;
+
+	/**
+	 * Private constructor om te voorkomen dat deze klasse (met static methoden)
+	 * aangemaakt kan worden.
+	 */
+	private PdfAttachmentUtil(List<CisConversionResult> cisConversionResultList) {
+		this.cisConversionResultList = cisConversionResultList;
+	}
+	
+	public PdfAttachmentUtil() {
+		
+	}
+
+	/**
+	 * Adds the given files in cisConversionResultList to the given rootPdf file.
+	 * 
+	 * @param cisConversionResultList
+	 * @param rootPdf
+	 * @throws IOException
+	 */
+	public static void addAttachmentInSinglePdf(List<CisConversionResult> cisConversionResultList)
+			throws IOException {
+		PdfAttachmentUtil pdfAttachmentUtil = new PdfAttachmentUtil(cisConversionResultList);
+		try {
+			pdfAttachmentUtil.addAttachmentInSinglePdf();
+		} finally {
+			pdfAttachmentUtil.finit();
+		}
+	}
+
+	protected void addAttachmentToPdf(CisConversionResult result, InputStream fileToAttach, String filename, String extension)
+			throws IOException {
+		PdfAttachmentUtil pdfAttachmentUtil = new PdfAttachmentUtil();
+		try (BufferedInputStream attachmentDocumentStream = new BufferedInputStream(fileToAttach)) {
+			pdfAttachmentUtil.addFileToPdf(attachmentDocumentStream, filename, extension, result);
+		} finally {
+			pdfAttachmentUtil.finit();
+		}
+	}
+
+	/**
+	 * Create a new pdf rootPdf based on the pdf in cisConversionResult and add all
+	 * files specified in cisConversionResult.attachments to it.
+	 * <p>
+	 * Note: Nothing is changed to the given cisConversionResult object and its
+	 * underlying files.
+	 * </p>
+	 * 
+	 * if there are no attachments null is returned otherwise rootPdf.
+	 * 
+	 * @param cisConversionResult
+	 *            wiht the given pdf and its attachments.
+	 * @param rootPdf
+	 *            the pdf created with the attachments embedded.
+	 * @throws IOException
+	 */
+	private void addAttachmentInSinglePdf() throws IOException {
+
+		for (CisConversionResult cisConversionResultAttachment : cisConversionResultList) {
+
+			if (cisConversionResultAttachment.getPdfResultFile() != null) {
+
+				try (InputStream attachmentDocumentStream = new BufferedInputStream(
+						new FileInputStream(cisConversionResultAttachment.getPdfResultFile()))) {
+
+					addFileToPdf(attachmentDocumentStream, cisConversionResultAttachment.getDocumentName(),
+							ConvertorUtil.PDF_FILETYPE, cisConversionResultAttachment);
+				}
+
+			} else {
+				LOGGER.debug("Skipping file because it is not available.");
+			}
+		}
+	}
+
+	private void finit() {
```

Comment:
What is the idea of finit? Is it 'finish' or 'f-initialize'? Please name it clearly